extensions/amp-video/0.1/test-e2e/test-amp-video-flexible-bitrate.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const VIEWPORT = {
+  HEIGHT: 768,
+  WIDTH: 1024,
+};
+
+const VIDEO_EVENTS = {
+  DOWNGRADE: 'd',
+  LOAD: 'l',
+  UNLOAD: 'u',
+};
+
+describes.endtoend(
+  'amp-video flexible bitrate',
+  {
+    fixture: 'amp-video/videos-cdn.html',
+    environments: ['single'],
+    experiments: ['flexible-bitrate'],
+    initialRect: {width: VIEWPORT.WIDTH, height: VIEWPORT.HEIGHT},
+  },
+  (env) => {
+    let controller;
+    let story;
+    let debugField;
+
+    beforeEach(async () => {
+      controller = env.controller;
+
+      story = await controller.findElement('amp-story.i-amphtml-story-loaded');
+      await controller.findElement('amp-story-page#page-1[active]');
+
+      debugField = await controller.findElement('input#flexible-bitrate-debug');
+    });
+
+    /**
+     * Tests for first load
+     */
+
+    it('when loading the page with good connection, the best quality source is played', async () => {
```

Comment:
please rephrase the tests using the `it` structure, it's always good to have the intent first and the conditions later, "it <intent (plays)> when <conditions (good connection)>"